torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -220,6 +220,15 @@ def _share_state_and_init_handle_attrs(
         attr_name_to_values[attr_name] = set()
     root_state._all_fsdp_states = traversal_utils._get_fsdp_states(root_module)
     root_state._all_handles = root_state._exec_order_data.all_handles  # share reference
+    # Update _has_optim_in_backward for each handle.
```

Comment:
Let us add up-front assertion that `use_orig_params=True` to use `_in_backward_optimizers`. Then, we can simply `assert root_state._use_orig_params` and then remove the `if flat_param._params is not None`.


(`if flat_param._params is not None` is currently being used as a proxy for `use_orig_params=True` or `False`.)

Reply:
I see! That makes sense.

I think it could be clearer still to check `if handle._use_orig_params` instead of directly checking `flat_param._params is not None` and leveraging that `handle._use_orig_params == True` implies `flat_param._params is not None` since whether `use_orig_params=True` or not is the source of truth.